Fixes for sign-in error handling

kevinansfield opened the pull request:
issue #5652, closes #5641

Two of the audit items were about errors showing when fields were empty and there not being enough space for validation messages.

Seeing as for the signup form we have a separate error message below the submit button I've tried to solve both issues by removing the inline messages and creating separate validation functions for sign-in and sign-up so that the relevant fields are still highlighted.

@JohnONolan, @ErisDS: Does this alternative process work for you or do we still want inline messages in some cases?

- removes inline errors for empty fields
- removes on-blur validation, instead only performing validation on submit
- separate validation routines for sign-in and forgot password
- highlight fields with errors when trying to submit
- adds a fix for "forgot" link overlapping long passwords and the new safari keychain icon (#5641)

TODO:
- [x] Update tests

JohnONolan commented:
Umm, I'm not sure if I'm understanding this correctly. It sounds like you've killed off inline validation completely, other than field-highlighting? If so, this definitely wasn't the desired outcome -- but if I'm getting it wrong please do set me straight

kevinansfield commented:
I killed the inline validation as on the sign-in page we also have the validation/error message at the bottom of the form and they weren't always in sync due to some validations running on blur which was confusing.

I've attached a video of how it works with this change. 
![zelda-signin-updates](https://cloud.githubusercontent.com/assets/415/9406396/1b417f70-47f8-11e5-9f18-c4ecca0fc98c.gif)

JohnONolan commented:
You know what, that looks totally awesome. Big :+1: the only thing I can see still missing is that the *default* icon colour is currently green. The default colour should be grey, it should only be green on successful validation.

You're totally right about losing inline validation here - this makes way more sense

kevinansfield commented:
I've added tests, an additional fix for #5641 and rebased.

@JohnONolan I'll probably tackle the default green icon issue in a separate PR as it's inherent to the way inline-validations works so likely something that will be happening app-wide.
